CTSK.py

```diff
@@ -0,0 +1,280 @@
+import os
+from typing import Tuple, Union, Type
+from abc import ABC, abstractmethod
+
+import numpy as np
+from Bio import SeqIO
+from Bio import SeqUtils
+
+
+class InvalidInput(ValueError):
+    """
+    Exception raised for invalid input values.
+
+    Inherits from ValueError.
+
+    Attributes:
+        message (str): Explanation of the error.
+    """
+    pass
+
+class NotImplementedError(ValueError):
+    """
+    Exception raised for call invalid class sample.
+
+    Inherits from ValueError.
+
+    Attributes:
+        message (str): Explanation of the error.
+    """
+    pass
+
+
+class BiologicalSequence(ABC, str):
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, sequence: str, alphabet: dict = None, complement_alphabet: dict = None):
+        self.sequence = None
+        self.alphabet = None
+        self.complement_alphabet = None
+
+
+    def setter(self, sequence: str, alphabet: dict, complement_alphabet: dict = None):
+        self.sequence = sequence
+        self.alphabet = alphabet
+        self.complement_alphabet = complement_alphabet
+
+    def __str__(self):
+        return (str(self.sequence))
+
+    def check_alphabet(self) -> bool:
+        """
+        Check if a sequence is DNA or RNA.
+
+        Returns:
+            bool: True if the sequence is DNA or RNA, False otherwise.
+        """
+
+        unique_chars = set(self.sequence)
+        if not (unique_chars <= self.alphabet):
```

Comment:
```suggestion
       return unique_chars <= self.alphabet
```
так чуть-чуть компактнее

Reply:
и сильно питонистее